cos-fleetshard-operator-debezium/pom.xml

```diff
@@ -64,6 +64,10 @@
             <groupId>org.bf2</groupId>
             <artifactId>cos-fleetshard-api</artifactId>
         </dependency>
+        <dependency>
+            <groupId>org.bf2</groupId>
+            <artifactId>cos-fleetshard-api-deployment</artifactId>
```

Comment:
is this required ?

Reply:
Yes. When you do a fresh checkout the dependency seems to be missing and the project doesn't compile.
I think when you delete your local `~/.m2/repository/org/bf2` cache then you can see it stops compiling. A `./mvnw clean package -am -pl $module_name` won't work as it won't build that dependent module.
If you have a better solution I can do that.

Reply:
I think this is an issue with quarkus but the issue here is that for quarkus pov, a `-deployment` dependency should not be part of the runtime

Reply:
@lburgazzoli really? So we call the module `-depl0yment` now, or how we gonna solve it?
On the other hand, it works when I install that module once and have it in my Maven cache.

Reply:
that is how quarkus works: the `-deployment` are used a build time and are not included in the runtime.
the dependency is specified by some metadata included in the `cos-fleetshard-api` module and resolved by the quarkus maven plugin.

see https://quarkus.io/guides/building-my-first-extension

Reply:
You mean the `cos-fleetshard` parent? Yeah, but often I don't want to build Camel when I only change Debezium. And I expect the build to work.
But back to the topic, can we keep it like this, or can I fix that dependency issue in a different way?

Reply:
> But back to the topic, can we keep it like this, or can I fix that dependency issue in a different way?

I would really like to avoid that as this has an impact on the produced artifacts and how quarkus works

Reply:
Agreed. But how? 🤔

Reply:
just remove the additional deps

you may need to rebuild the project instead of individual module till we find a solution :)

Reply:
Ha! Found the solution!